Src/tim.c

```diff
@@ -53,9 +53,9 @@ void MX_TIM16_Init(void)
   TIM_BreakDeadTimeConfigTypeDef sBreakDeadTimeConfig;
 
   htim16.Instance = TIM16;
-  htim16.Init.Prescaler = 0;
+  htim16.Init.Prescaler = 15; //(timer Frequency/(PWM STEPS * PWM FREQUENCY))-1
```

Comment:
As it turns out, this prescaler just barely works for the purpose of generating 50 Hz pwm. (prescaler of 15 gets us a 3Mhz effective clock rate from the 48MHz input clock, which means the 16 bit timer can count up to 0.0218s before overflowing).

You expressed doubt about the prescaler though, so you might like to read about the prescale register and how it works. (17.6.10 in the reference manual)